### PR TITLE
[9.0.0] Merge master into 9.0.0-dev

### PR DESCRIPTION
**Pull request by sunshinejr**
I merged changes from master into 9.0.0-dev, but there were some conflicts and I would love if you guys could take a look at it and make sure everything is okay :D

> Conflicts:
> - Cartfile.resolved
> - Demo/Podfile.lock
> -	Package.swift
> -	Tests/ErrorTests.swift
> -	Tests/MoyaProviderIntegrationTests.swift
> -	Tests/RxSwiftMoyaProviderTests.swift
> -	Tests/TestHelpers.swift
> -	docs/Examples/AlamofireValidation.md

### CODE REVIEW

**Comment by pedrovereza** [comment]
@sunshinejr Last time we rebased `9.0.0-dev` with `master`, which I feel may be easier 🤔

**Comment by sunshinejr** [comment]
@pedrovereza Yeah, I did that as well, but didn't push to the branch itself, but created a new one and did a PR since it was a big merge and wanted to make sure I didn't break anything. Also, this problem with Changelog is strange since there was no conflict in it 🤔

File: Package.swift [hunk 1]
```diff
@@ -16,10 +16,10 @@ let package = Package(
         )
     ],
     dependencies: [
-        .Package(url: "https://github.com/Alamofire/Alamofire", majorVersion: 4),
-        .Package(url: "https://github.com/ReactiveCocoa/ReactiveSwift", majorVersion: 1),
-        .Package(url: "https://github.com/ReactiveX/RxSwift", majorVersion: 3, minor: 3),
-        .Package(url: "https://github.com/antitypical/Result", majorVersion: 3),
+        .Package(url: "https://github.com/Alamofire/Alamofire.git", majorVersion: 4),
+        .Package(url: "https://github.com/ReactiveCocoa/ReactiveSwift.git", majorVersion: 1),
+        .Package(url: "https://github.com/ReactiveX/RxSwift.git", majorVersion: 3),
```
Comment (pedrovereza) [inline Package.swift]: I think we need RxSwift to be 3.3 because of #1020 🤔

**Comment by pedrovereza** [comment]
@sunshinejr Oh I see, let's go ahead and just make a commit fixing the changelog then 😄

File: Package.swift (hunk 1 above)
Comment (sunshinejr) [inline Package.swift]: Nice catch!

**Comment by sunshinejr** [comment]
Fixed `Changelog.md` & `Package.swift` - thanks @pedrovereza! Also, I've checked once again and it _seems alright_, but another look wouldn't hurt :D

**Pull request merged**